### PR TITLE
Cast function pointers legally.

### PR DESCRIPTION
**Pull request by marsupial**

### CODE REVIEW

File: include/cling/Utils/Casting.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    ///\brief Legally cast a void* to a function.
+    ///
+    /// \param [in] The void* to cast
+    ///
+    /// \return The function's address.
+    ///
+    template <class T>
+    T VoidToFunction(void *ptr) {
+      return UintToFunction<T>(static_cast<void*>(ptr));
```
Comment (karies) [inline include/cling/Utils/Casting.h]: Why do you need to `static_cast` a `void*` to a `void*`? Did you mean `static_cast<uintptr_t>`?

File: include/cling/Utils/Casting.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  namespace utils {
+
+    ///\brief Legally cast a pointer to a function to void*.
+    ///
+    /// \param [in] The function to cast
+    ///
+    /// \return A void* of the functions address.
+    ///
+    template <class T>
+    void* FunctionToVoid(T *fPtr) {
```
Comment (karies) [inline include/cling/Utils/Casting.h]: Can we throw in `Ptr`? "Void" Without "Ptr" sounds weird, but then we also need it for Function, so I guess `FunctionPtrToVoidPtr`? Or `FunPtrToVoidPtr` is fine, too.

Comment (karies) [inline include/cling/Utils/Casting.h]: Could you rename `fPtr` to e.g. `funptr`? Else it's an unfortunate clash with ROOT's coding conventions - yes, not relevant for cling, but that change would still be much appreciated :-)

**Comment by marsupial** [comment]
Sure, was over confident and had many a mistake on the way.
Could you go into Travis and cancel all of this PRs builds prior to the last?

**Comment by karies** [comment]
My tests pass - good enough for me. Thanks for the really nice PR, merged!

**Pull request closed without merging**